dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/OrgUnitMergeController.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2004-2021, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import lombok.AllArgsConstructor;
+
+import org.hisp.dhis.common.DhisApiVersion;
+import org.hisp.dhis.merge.orgunit.OrgUnitMergeQuery;
+import org.hisp.dhis.merge.orgunit.OrgUnitMergeService;
+import org.hisp.dhis.webapi.mvc.annotation.ApiVersion;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Org unit merge API controller.
+ *
+ * @author Lars Helge Overland
+ */
+@RestController
+@AllArgsConstructor
+@ApiVersion( { DhisApiVersion.DEFAULT, DhisApiVersion.ALL } )
+@PreAuthorize( "hasRole('ALL') or hasRole('F_ORGANISATION_UNIT_MERGE')" )
+public class OrgUnitMergeController
+{
+    private final OrgUnitMergeService service;
+
+    @PostMapping( value = "/merge/organisationUnits", produces = { APPLICATION_JSON_VALUE } )
```

Comment:
Wouldn't it be more natural the other way around: `/organisationUnits/merge`? Since we have other operations being `/organisationUnits/...`

Reply:
Yeah I discussed that with myself, pros and cons, one drawback is that we use the same API path as `OrganisationUnitController` and might conflict with it (`/api/organisationUnits/<uid>`). We might also have merge of data elements and category options in the future.

Reply:
I am happy to change it though, since this is POST it won't conflict with GET methods.